### DISCUSSION TITLE
Integrating Hades Workshop with Memoria (Details in description) - Allowing an editor mode option instead of just Play

### DISCUSSION THREAD

**Discussion by WarpedEdge** [Albeoris/Memoria#231] [Ideas]
Spoke with Tirlititi and wanted to open a discussion on this to hopefully get some people working together on this/breed some discussion in general for how well this could play to the future of Memoria and modding of FFIX. The main points that will be needed to do this are below.

1) Porting the whole code of Hades Workshop from C++ into C#
2) Add a way to switch to an "editor mode" instead of the "play the game mode" (this could ease a lot of things)
3) Update the features of the editor mode to keep up with the features of the Memoria engine (definitely required)

Is anyone interested in something like this? If so, is anyone interested in assisting with this project? I am learning more and more code each day (mainly Python unfortunately Haha), but I am also extremely detailed in documentation processes, guides, etc. which could bring in a lot of new people if a guide is written from top to bottom.

**Comment by stir472** [comment]
Hello, I would love to be able to help in any way, but unfortunately I am not a programmer and I do not understand code, but if you know of any way that I can help you, be happy to do so.

**Comment by WarpedEdge** [comment]
Speaking with Tirlititi the HWS code would need to be entirely re-written
from C++ to C#. If it can get to that point I believe he would try and work
on it again to integrate it with Memoria which would be amazing, but as it
stands, he is the only one who has done anything with the tool and does not
have enough time. If you know someone who can do that, that would be the
best place to start as I am sure he would provide guidance from that point.

**Reply by Albeoris** [reply]
@WarpedEdge , [there](https://github.com/Albeoris/EvilForest) is a small peace of functionality that HW can provide, so if you want, you can start here.

Projects like HW take decades to write, and when it comes to “rewriting from scratch,” most often no one has enough time or motivation to complete this titanic work.

Instead of "rewriting the project", I recommend taking a small task that solves one specific problem and starting from there.

For example, here's the problem:
* https://github.com/Albeoris/Memoria/issues/403

To solve this, you don’t need to rewrite the entire HW in C#; it’s enough to write a small module that can then be integrated into HW and/or Memoria.

I recommend focusing on solving small problems like this, as you will see the results of your efforts and receive instant feedback that will motivate you to continue.

**Comment by WarpedEdge** [comment]
@Albeoris Thanks for the feedback. Unfortunately I do not have modding
abilities like you awesome folk. I am better at writing guides and
supplemental material for other people or new users that come along...which
i was supposed to work on for FFIX but real life got me busy and I am also
writing a guide for FFT Colosseum :L
Anyways, I will take a look into this anyways, see if maybe I can learn
enough to work this part, if no one else gets to it before me :). One
question, as I am learning this, is it better for me to focus on a
particular language? I am at the moment learning python, not C++ or C#.

**Reply by Albeoris** [reply]
@WarpedEdge `Python` is an universal solution in the industry. `C#` is a best language for gamedev and for beginners because of well-documented framework. `Go` and `Rust` are popular in certain segments. But avoid `C++` because you need 10+ years to get how to write safe and maintainable code on it. :)

**Comment by WarpedEdge** [comment]
Understood. Thank you for the insight. Ill learn more of Python first as I
am trying to move from Site engineer role to more DevOps so i need it haha.
On my own time ill learn C# or Rust as i heard its coming up in the world.
Also welcome back! Happy to see your name appearing again and awesome stuff
on EvilForest if only I could fully comprehend its capabilities.
